rotkehlchen/tests/unit/test_data_updates.py

```diff
@@ -90,16 +113,55 @@ def test_no_update_performed(data_updater: RotkiDataUpdater) -> None:
     """
     Check that the updates don't execute if we have a higher version locally
     """
-    # set a higer last version applied
+    # set a higher last version applied
     with data_updater.user_db.conn.write_ctx() as write_cursor:
-        write_cursor.execute(
+        write_cursor.executemany(
             'INSERT OR REPLACE INTO settings(name, value) VALUES (?, ?)',
-            (UpdateType.SPAM_ASSETS.serialize(), 999),
+            [
+                (UpdateType.SPAM_ASSETS.serialize(), 999),
+                (UpdateType.RPC_NODES.serialize(), 999),
+            ],
         )
 
     with (
         patch('requests.get', wraps=mock_github_data_response_old_update),
         patch.object(data_updater, 'update_spam_assets') as spam_assets,
+        patch.object(data_updater, 'update_rpc_nodes') as rpc_nodes,
     ):
         data_updater.check_for_updates()
         assert spam_assets.call_count == 0
+        assert rpc_nodes.call_count == 0
+
+
+def test_update_rpc_nodes(data_updater: RotkiDataUpdater) -> None:
+    """Test that rpc nodes for different blockchains are updated correctly.."""
+    data_updater.msg_aggregator.consume_warnings()
+
+    # check db state of the default rpc nodes before updating
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        cursor.execute('SELECT COUNT(*) FROM default_rpc_nodes')
+        assert cursor.fetchone()[0] == 10
+
+    # check the db state of the user's rpc_nodes
+    with data_updater.user_db.conn.read_ctx() as cursor:
+        cursor.execute('SELECT COUNT(*) FROM rpc_nodes')
+        assert cursor.fetchone()[0] == 10
+
+    with patch('requests.get', wraps=mock_github_data_response):
+        data_updater.check_for_updates()
+
+    # check the db state after updating
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        cursor.execute('SELECT COUNT(*) FROM default_rpc_nodes')
+        assert cursor.fetchone()[0] == 2
+
+    # check that 2 nodes are present in the user db
+    # 9 nodes were deleted since the updated rpc nodes data
+    # did not contain them.
+    with data_updater.user_db.conn.read_ctx() as cursor:
```

Comment:
can you also add a custom user node to the user DB and make sure this is not deleted? I think from the code it should not but need to have it in the test